### PR TITLE
Completed Cart Functionality

### PR DESCRIPTION
**Pull request by nareshpoonia**
Completed Cart Functionality

### CODE REVIEW

File: src/components/cart-products.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import { useCartContext } from "../context/cart-context";
+import { useState } from "react";
+
+function Cartproducts(props) {
+  const { cartState, cartDispatch } = useCartContext();
+
+  console.log(cartState.cartItem);
```
Comment (theprogrammerinyou) [inline src/components/cart-products.js]: Do, remove this log while merging.

**Review by theprogrammerinyou — APPROVED** [review]
Do remove those logs, as part of cleaning code, otherwise code looks good.Thank you.

File: src/context/cart-context.js [hunk 2]
```diff
@@ -1,16 +1,63 @@
...
+          if (item._id === action.payload && item.quantity > 1) {
+            return { ...item, quantity: item.quantity - 1 };
+          }
+          return item;
+        }),
+      };
+    case "REMOVE_FROM_CART":
+      return {
+        ...state,
+        cartArray: state.cartArray.map((item) => {
```
Comment (shivampandey0) [inline src/context/cart-context.js]: use `Array.filter() `to remove item from list

**Review by shivampandey0 — COMMENTED** [review]
Please fix the suggested changes mentioned over call.

File: src/components/cart-products.js (hunk 1 above)
Comment (nareshpoonia) [reply src/components/cart-products.js]: Removed console.log, Thanks for pointing out

File: src/context/cart-context.js (hunk 2 above)
Comment (nareshpoonia) [reply src/context/cart-context.js]: Used Array.filter instead, thanks for suggesting

**Pull request merged**